src/components/layouts/navbar.tsx

```diff
@@ -0,0 +1,177 @@
+"use client";
+
+import Link from "next/link";
+import { RxHamburgerMenu } from "react-icons/rx";
+import { IoIosClose } from "react-icons/io";
+import Image from "next/image";
+import useMainStore from "@/hooks/main/store";
+import { MenuContent, MenuItem, MenuRoot, MenuTrigger } from "../ui/menu";
+import { motion, AnimatePresence } from "framer-motion";
+import { Box, IconButton } from "@chakra-ui/react";
+import clsx from "clsx";
+import useAuthStore from "@/contexts/auth/store";
+import { Button } from "../ui/button";
+import { BsPerson } from "react-icons/bs";
+import { LiaSignOutAltSolid } from "react-icons/lia";
+import useRoleDialogStore from "@/hooks/roleDialog/store";
+import {
+  useHideNavbarOnScroll,
+  useIsNavbarActive,
+} from "@/common/utils/navbar";
+import { slideVariants } from "@/common/types/animationVariants";
+
+const navLinks = [
+  { href: "/vacancy", label: "Vacancy" },
+  { href: "/history", label: "History" },
+  { href: "/notification", label: "Notification" },
+];
+
+type NavLinkProps = {
+  href: string;
+  label: string;
+};
+
+const NavLink = ({ href, label }: NavLinkProps) => (
+  <Link href={href}>
+    <Button
+      className={clsx(
+        "px-3 py-2 rounded-md",
+        useIsNavbarActive(href)
+          ? "bg-gray-500 text-blue-300 font-semibold"
+          : "hover:text-blue-400 hover:bg-gray-500 transition-all duration-200",
+      )}
+      disabled={useIsNavbarActive(href)}
+    >
+      {label}
+    </Button>
+  </Link>
+);
+
+const MobileNavLink = ({ href, label }: NavLinkProps) => (
+  <Link href={href}>
+    <Button
+      className={clsx(
+        "w-full px-3 py-2 rounded-md",
+        useIsNavbarActive(href)
+          ? "bg-gray-500 text-blue-300 font-semibold"
+          : "hover:text-blue-400 hover:bg-gray-500 transition-all duration-200",
+      )}
+      disabled={useIsNavbarActive(href)}
+    >
+      {label}
+    </Button>
+  </Link>
+);
+
+const Navbar = () => {
+  const { isLogin, role, signOut } = useAuthStore();
+  const { isNavigationOpen, setIsNavigationOpen, setIsLoginDialogOpen } =
```

Comment:
for this state naming, i think it better to be like this:
~setIsNavigationOpen~ setNavigationOpen
~setIsLoginDialogOpen~ setLoginDialogOpen